app/src/main/java/com/frinder/frinder/activity/DiscoverActivity.java

```diff
@@ -0,0 +1,139 @@
+
+import android.location.Location;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.Toolbar;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.widget.Toast;
+
+import com.facebook.Profile;
+import com.frinder.frinder.R;
+import com.frinder.frinder.dataaccess.UserFirebaseDas;
+import com.frinder.frinder.model.User;
+
+import java.util.ArrayList;
+import java.util.TreeSet;
+
+public class DiscoverActivity extends AppCompatActivity implements UserFirebaseDas.UserDasInterface{
+    private static final String TAG = "DiscoverActivity";
+    ArrayList<User> users;
+    DiscoverUsersAdapter adapter;
+    Profile profile;
+    User currentUser;
+    UserFirebaseDas userFirebaseDas;
+
+    //ToDo Mallika - Remove this constant when filters/settings screen is ready
+    //Assuming that we are looking for people in a radius of 150m which is about 574.147ft.
+    private final static int SEARCH_RADIUS = 175; //unit is meters
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_discover);
+
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+
+        userFirebaseDas = new UserFirebaseDas(DiscoverActivity.this);
+
+        profile = Profile.getCurrentProfile();
+        userFirebaseDas.getUser(profile.getId());
+
+        RecyclerView rvDiscoverusers = (RecyclerView) findViewById(R.id.rvDiscoverUsers);
+        users = new ArrayList<>();
+        adapter = new DiscoverUsersAdapter(this, users);
+        rvDiscoverusers.setAdapter(adapter);
+        rvDiscoverusers.setLayoutManager(new LinearLayoutManager(this));
+    }
+
+    public void getNearbyUsers() {
+        //Clear the list each time discover menu button is clicked as user list will change based on who is nearby
+        users.clear();
+        adapter.notifyDataSetChanged();
+        // ToDo Mallika - change userFirebaseDas.getAllUsers() to return users based on current filters
+        userFirebaseDas.getAllUsers();
+    }
+
+    @Override
+    public void readAllUsersComplete(ArrayList<User> userList) {
+        Log.d(TAG, "in readAllUsersComplete");
+        Log.d(TAG, userList.toString());
+
+        //ToDo - transfer this logic to Firebase if possible
+
+        ArrayList<User> nearbyUsers = new ArrayList<>();
+        ArrayList<Float> distanceFromCurrentUser = new ArrayList<>();
+        TreeSet<Float> distances = new TreeSet<>();
+
+        for (User user : userList) {
+            if (!user.getUid().contentEquals(currentUser.getUid())) {
+                float[] results = new float[1];
+                Location.distanceBetween(currentUser.getLocation()[0], currentUser.getLocation()[1],
```

Comment:
@mallikaviswas this operation is fine for lesser number of users but imagine with 1000s of user near by this could be expensive operation, we could explore the geofire in the next stage where firebase would just give the nearby user list rather than giving the full list and we filtering it by distance in the app.

Reply:
Yes you're right Sanal. This is just until we move it to Firebase. I should make sure I put a TODO there.